### ISSUE TITLE
[Bug Report]: Cannot run and build with dynamic `require` statement

### ISSUE THREAD

**Issue by CrazyboyQCD** [farm-fe/farm#941]
### Steps to reproduce
To be minimum.
1. Initialize a nake project with one index.html and one index.js.
2. Add `farm` and `component-classes`.
3. Try to use `component-classes` to avoid tree-shaking.
![image](https://github.com/farm-fe/farm/assets/53971641/5ae73a58-0a17-404a-b010-37b4a3ac3dbe)

The error was caused by this snippet from `component-classes`.
![image](https://github.com/farm-fe/farm/assets/53971641/de90a609-9263-47a3-baf5-48e294efd069)

### What is expected?

Running and building with no errors.

### What is actually happening?

Running and building with errors.
Runing:
![image](https://github.com/farm-fe/farm/assets/53971641/70b28320-e768-484f-ad14-80474562876d)
Building:
![image](https://github.com/farm-fe/farm/assets/53971641/88364094-c991-4de1-b321-941678dea0b9)

### System Info

```Shell
System:
    OS: Windows 11 10.0.22000
  Binaries:
    Node: 18.19.0 - D:\nodejs\node.EXE
    npm: 10.2.3 - D:\nodejs\npm.CMD
    pnpm: 8.14.1 - D:\nodejs\pnpm.CMD
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.22000.120.0), Chromium (121.0.2277.83)
```

### Any additional comments?

Since I'm migrating from an old project, the `component-classes` dep cannot be removed in short term.

**Comment by wre232114** [comment]
Thanks for reporting issues. I'll take a look.

**Comment by wre232114** [comment]
fixed in `@farmfe/core@0.16.4`
